transient test failure in BrokerServerViewTest.testMultipleServerAddedRemovedSegment

In https://travis-ci.org/druid-io/druid/jobs/70802104

org.apache.zookeeper.KeeperException$NodeExistsException: KeeperErrorCode = NodeExists for /druid/segments/locahost:0

	at org.apache.zookeeper.KeeperException.create(KeeperException.java:119)

	at org.apache.zookeeper.KeeperException.create(KeeperException.java:51)

	at org.apache.zookeeper.ZooKeeper.create(ZooKeeper.java:783)

	at org.apache.curator.framework.imps.CreateBuilderImpl$11.call(CreateBuilderImpl.java:698)

	at org.apache.curator.framework.imps.CreateBuilderImpl$11.call(CreateBuilderImpl.java:675)

	at org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:107)

	at org.apache.curator.framework.imps.CreateBuilderImpl.pathInForeground(CreateBuilderImpl.java:671)

	at org.apache.curator.framework.imps.CreateBuilderImpl.protectedPathInForeground(CreateBuilderImpl.java:453)

	at org.apache.curator.framework.imps.CreateBuilderImpl.forPath(CreateBuilderImpl.java:443)

	at org.apache.curator.framework.imps.CreateBuilderImpl.forPath(CreateBuilderImpl.java:423)

	at org.apache.curator.framework.imps.CreateBuilderImpl$4.forPath(CreateBuilderImpl.java:331)

	at org.apache.curator.framework.imps.CreateBuilderImpl$4.forPath(CreateBuilderImpl.java:267)

	at io.druid.client.BrokerServerViewTest.setupZNodeForServer(BrokerServerViewTest.java:382)

	at io.druid.client.BrokerServerViewTest.testMultipleServerAddedRemovedSegment(BrokerServerViewTest.java:183)d

another failure in same
```
io.druid.client.BrokerServerViewTest.testMultipleServerAddedRemovedSegment
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at io.druid.client.BrokerServerViewTest.__CLR4_0_0h9p6yz82e(BrokerServerViewTest.java:206)
	at io.druid.client.BrokerServerViewTest.testMultipleServerAddedRemovedSegment(BrokerServerViewTest.java:154)
```

Log
```
2015-07-29T03:33:47,496 INFO [Thread-10] org.apache.zookeeper.server.ZooKeeperServerMain - Starting server
2015-07-29T03:33:47,496 INFO [Thread-10] org.apache.zookeeper.server.ZooKeeperServer - tickTime set to 3000
2015-07-29T03:33:47,496 INFO [Thread-10] org.apache.zookeeper.server.ZooKeeperServer - minSessionTimeout set to -1
2015-07-29T03:33:47,496 INFO [Thread-10] org.apache.zookeeper.server.ZooKeeperServer - maxSessionTimeout set to -1
2015-07-29T03:33:47,497 INFO [Thread-10] org.apache.zookeeper.server.NIOServerCnxnFactory - binding to port 0.0.0.0/0.0.0.0:44924
2015-07-29T03:33:48,499 WARN [main] org.apache.curator.CuratorZookeeperClient - session timeout [2000] is less than connection timeout [10000]
2015-07-29T03:33:48,500 INFO [main] org.apache.curator.framework.imps.CuratorFrameworkImpl - Starting
2015-07-29T03:33:48,500 INFO [main] org.apache.zookeeper.ZooKeeper - Initiating client connection, connectString=127.0.0.1:44924 sessionTimeout=2000 watcher=org.apache.curator.ConnectionState@5eed0dcc
2015-07-29T03:33:48,501 INFO [main-SendThread(127.0.0.1:44924)] org.apache.zookeeper.ClientCnxn - Opening socket connection to server 127.0.0.1/127.0.0.1:44924. Will not attempt to authenticate using SASL (unknown error)
2015-07-29T03:33:48,502 INFO [main-SendThread(127.0.0.1:44924)] org.apache.zookeeper.ClientCnxn - Socket connection established to 127.0.0.1/127.0.0.1:44924, initiating session
2015-07-29T03:33:48,502 INFO [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:44924] org.apache.zookeeper.server.NIOServerCnxnFactory - Accepted socket connection from /127.0.0.1:56444
2015-07-29T03:33:48,503 INFO [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:44924] org.apache.zookeeper.server.ZooKeeperServer - Client attempting to establish new session at /127.0.0.1:56444
2015-07-29T03:33:48,504 INFO [SyncThread:0] org.apache.zookeeper.server.persistence.FileTxnLog - Creating new log file: log.1
2015-07-29T03:33:48,506 INFO [SyncThread:0] org.apache.zookeeper.server.ZooKeeperServer - Established session 0x14ed7df876a0000 with negotiated timeout 6000 for client /127.0.0.1:56444
2015-07-29T03:33:48,506 INFO [main-SendThread(127.0.0.1:44924)] org.apache.zookeeper.ClientCnxn - Session establishment complete on server 127.0.0.1/127.0.0.1:44924, sessionid = 0x14ed7df876a0000, negotiated timeout = 6000
2015-07-29T03:33:48,507 INFO [main-EventThread] org.apache.curator.framework.state.ConnectionStateManager - State change: CONNECTED
2015-07-29T03:33:48,514 INFO [ServerInventoryView-0] io.druid.client.BatchServerInventoryView - Inventory Initialized
2015-07-29T03:33:48,520 INFO [ProcessThread(sid:0 cport:-1):] org.apache.zookeeper.server.PrepRequestProcessor - Got user-level KeeperException when processing sessionid:0x14ed7df876a0000 type:create cxid:0x8 zxid:0x5 txntype:-1 reqpath:n/a Error Path:/druid/segments Error:KeeperErrorCode = NoNode for /druid/segments
2015-07-29T03:33:48,542 INFO [ServerInventoryView-0] io.druid.curator.inventory.CuratorInventoryManager - Created new InventoryCacheListener for /druid/segments/locahost:0
2015-07-29T03:33:48,542 INFO [ServerInventoryView-0] io.druid.client.BatchServerInventoryView - New Server[DruidServerMetadata{name='locahost:0', host='locahost:0', maxSize=10000000, tier='default_tier', type='historical', priority='0'}]
2015-07-29T03:33:48,548 INFO [ServerInventoryView-0] io.druid.curator.inventory.CuratorInventoryManager - Created new InventoryCacheListener for /druid/segments/localhost:4
2015-07-29T03:33:48,548 INFO [ServerInventoryView-0] io.druid.client.BatchServerInventoryView - New Server[DruidServerMetadata{name='localhost:4', host='localhost:4', maxSize=10000000, tier='default_tier', type='historical', priority='0'}]
2015-07-29T03:33:48,549 INFO [ServerInventoryView-0] io.druid.curator.inventory.CuratorInventoryManager - Created new InventoryCacheListener for /druid/segments/localhost:3
2015-07-29T03:33:48,550 INFO [ServerInventoryView-0] io.druid.client.BatchServerInventoryView - New Server[DruidServerMetadata{name='localhost:3', host='localhost:3', maxSize=10000000, tier='default_tier', type='historical', priority='0'}]
2015-07-29T03:33:48,551 INFO [ServerInventoryView-0] io.druid.curator.inventory.CuratorInventoryManager - Created new InventoryCacheListener for /druid/segments/localhost:2
2015-07-29T03:33:48,551 INFO [ServerInventoryView-0] io.druid.client.BatchServerInventoryView - New Server[DruidServerMetadata{name='localhost:2', host='localhost:2', maxSize=10000000, tier='default_tier', type='historical', priority='0'}]
2015-07-29T03:33:48,553 INFO [ServerInventoryView-0] io.druid.curator.inventory.CuratorInventoryManager - Created new InventoryCacheListener for /druid/segments/localhost:1
2015-07-29T03:33:48,553 INFO [ServerInventoryView-0] io.druid.client.BatchServerInventoryView - New Server[DruidServerMetadata{name='localhost:1', host='localhost:1', maxSize=10000000, tier='default_tier', type='historical', priority='0'}]
2015-07-29T03:37:58,552 INFO [ProcessThread(sid:0 cport:-1):] org.apache.zookeeper.server.PrepRequestProcessor - Processed session termination for sessionid: 0x14ed7df876a0000
2015-07-29T03:37:58,554 INFO [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:44924] org.apache.zookeeper.server.NIOServerCnxn - Closed socket connection for client /127.0.0.1:56444 which had sessionid 0x14ed7df876a0000
2015-07-29T03:37:58,554 INFO [main] org.apache.zookeeper.ZooKeeper - Session: 0x14ed7df876a0000 closed
2015-07-29T03:37:58,554 INFO [main-EventThread] org.apache.zookeeper.ClientCnxn - EventThread shut down
2015-07-29T03:37:58,555 INFO [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:44924] org.apache.zookeeper.server.NIOServerCnxnFactory - NIOServerCnxn factory exited run method
2015-07-29T03:37:58,555 INFO [Thread-10] org.apache.zookeeper.server.ZooKeeperServer - shutting down
2015-07-29T03:37:58,555 INFO [main] org.apache.zookeeper.server.ZooKeeperServer - shutting down
2015-07-29T03:37:58,555 INFO [Thread-10] org.apache.zookeeper.server.SessionTrackerImpl - Shutting down
2015-07-29T03:37:58,555 INFO [main] org.apache.zookeeper.server.SessionTrackerImpl - Shutting down
2015-07-29T03:37:58,555 INFO [Thread-10] org.apache.zookeeper.server.PrepRequestProcessor - Shutting down
2015-07-29T03:37:58,555 INFO [main] org.apache.zookeeper.server.PrepRequestProcessor - Shutting down
2015-07-29T03:37:58,555 INFO [Thread-10] org.apache.zookeeper.server.SyncRequestProcessor - Shutting down
2015-07-29T03:37:58,555 INFO [ProcessThread(sid:0 cport:-1):] org.apache.zookeeper.server.PrepRequestProcessor - PrepRequestProcessor exited loop!
2015-07-29T03:37:58,555 INFO [main] org.apache.zookeeper.server.SyncRequestProcessor - Shutting down
2015-07-29T03:37:58,555 INFO [SyncThread:0] org.apache.zookeeper.server.SyncRequestProcessor - SyncRequestProcessor exited!
2015-07-29T03:37:58,555 INFO [main] org.apache.zookeeper.server.FinalRequestProcessor - shutdown of request processor complete
2015-07-29T03:37:58,555 INFO [Thread-10] org.apache.zookeeper.server.FinalRequestProcessor - shutdown of request processor complete

```

--- a/server/src/test/java/io/druid/client/BrokerServerViewTest.java
+++ b/server/src/test/java/io/druid/client/BrokerServerViewTest.java
@@ -371,15 +371,24 @@ public class BrokerServerViewTest extends CuratorTestBase
 
   private void setupZNodeForServer(DruidServer server) throws Exception
   {
+    final String zNodePathAnnounce = ZKPaths.makePath(announcementsPath, server.getHost());
+    final String zNodePathSegment = ZKPaths.makePath(inventoryPath, server.getHost());
+
+    if (curator.checkExists().forPath(zNodePathAnnounce) != null) {
+      curator.delete().guaranteed().forPath(zNodePathAnnounce);
+    }
+    if (curator.checkExists().forPath(zNodePathSegment) != null) {
+      curator.delete().guaranteed().forPath(zNodePathSegment);
+    }
     curator.create()
            .creatingParentsIfNeeded()
            .forPath(
-               ZKPaths.makePath(announcementsPath, server.getHost()),
+               zNodePathAnnounce,
                jsonMapper.writeValueAsBytes(server.getMetadata())
            );
     curator.create()
            .creatingParentsIfNeeded()
-           .forPath(ZKPaths.makePath(inventoryPath, server.getHost()));
+           .forPath(zNodePathSegment);
   }
 
   private DataSegment dataSegmentWithIntervalAndVersion(String intervalStr, String version)

